tfsdk/serve_resource_config_validators_test.go

```diff
@@ -0,0 +1,96 @@
+package tfsdk
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type testServeResourceTypeConfigValidators struct{}
+
+func (dt testServeResourceTypeConfigValidators) GetSchema(_ context.Context) (Schema, []*tfprotov6.Diagnostic) {
+	return Schema{
+		Attributes: map[string]Attribute{
+			"string": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+		},
+	}, nil
+}
+
+func (dt testServeResourceTypeConfigValidators) NewResource(_ context.Context, p Provider) (Resource, []*tfprotov6.Diagnostic) {
+	provider, ok := p.(*testServeProvider)
+	if !ok {
+		prov, ok := p.(*testServeProviderWithMetaSchema)
+		if !ok {
+			panic(fmt.Sprintf("unexpected provider type %T", p))
+		}
+		provider = prov.testServeProvider
+	}
+	return testServeResourceConfigValidators{
+		provider: provider,
+	}, nil
+}
+
+var testServeResourceTypeConfigValidatorsSchema = &tfprotov6.Schema{
+	Block: &tfprotov6.SchemaBlock{
+		Attributes: []*tfprotov6.SchemaAttribute{
+			{
+				Name:     "string",
+				Optional: true,
+				Type:     tftypes.String,
+			},
+		},
+	},
+}
+
+var testServeResourceTypeConfigValidatorsType = tftypes.Object{
+	AttributeTypes: map[string]tftypes.Type{
+		"string": tftypes.String,
+	},
+}
+
+type testServeResourceConfigValidators struct {
+	provider *testServeProvider
+}
+
+func (r testServeResourceConfigValidators) Create(ctx context.Context, req CreateResourceRequest, resp *CreateResourceResponse) {
+}
+func (r testServeResourceConfigValidators) Read(ctx context.Context, req ReadResourceRequest, resp *ReadResourceResponse) {
+}
+func (r testServeResourceConfigValidators) Update(ctx context.Context, req UpdateResourceRequest, resp *UpdateResourceResponse) {
+}
+func (r testServeResourceConfigValidators) Delete(ctx context.Context, req DeleteResourceRequest, resp *DeleteResourceResponse) {
+}
+
+func (r testServeResourceConfigValidators) ConfigValidators(ctx context.Context) []ResourceConfigValidator {
+	r.provider.validateResourceConfigCalledResourceType = "test_config_validators"
+
+	return []ResourceConfigValidator{
+		newTestResourceConfigValidator(r.provider.validateResourceConfigImpl),
```

Comment:
Do we want to test having multiple validators? 🤔

Reply:
I'll follow this up with one last commit to cover that 👍